### PR TITLE
fix(@angular/cli): enable full node module resolution for project deps

### LINKED ISSUE #6248: Webpack config not finding nested node_modules
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.

```
@angular/cli: 1.0.2
node: 6.10.1
os: darwin x64
@angular/common: 4.1.1
@angular/compiler: 4.1.1
@angular/core: 4.1.1
@angular/forms: 4.1.1
@angular/http: 4.1.1
@angular/platform-browser: 4.1.1
@angular/platform-browser-dynamic: 4.1.1
@angular/router: 4.1.1
@angular/cli: 1.0.2
@angular/compiler-cli: 4.1.1
```

### Repro steps.

1. Clone the repo: https://github.com/jshcrowthe/ng-cli-nested-node-modules.git
1. Enter the repo dir
1. Run `npm install`
1. Run `ng serve` 
1. Observe error from `ng serve` command

### The log given by the failure.

```log
ERROR in ./~/firebase/app/shared_promise.js
Module not found: Error: Can't resolve 'promise-polyfill' in '$PATH_TO_REPO/nested-node-module/node_modules/firebase/app'
 @ ./~/firebase/app/shared_promise.js 22:35-62
 @ ./~/firebase/app/firebase_app.js
 @ ./~/firebase/app.js
 @ ./src/app/app.component.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
```

### Desired functionality.
Webpack should compile this module and it's nested children correctly

### Mention any other details that might be useful.
n/a

### PR DESCRIPTION
**Pull request by clydin**
As discussed in issue #6248.

Closes #6248

### CODE REVIEW

File: packages/@angular/cli/models/webpack-configs/common.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export function getCommonConfig(wco: WebpackConfigOptions) {
     devtool: buildOptions.sourcemaps ? 'source-map' : false,
     resolve: {
       extensions: ['.ts', '.js'],
-      modules: [nodeModules],
+      modules: ['node_modules', nodeModules],
```
Comment (filipesilva) [inline packages/@angular/cli/models/webpack-configs/common.ts]: What is the difference between `modules: ['node_modules', nodeModules],` and just `modules: ['node_modules']`?

Comment (clydin) [inline packages/@angular/cli/models/webpack-configs/common.ts]: It provides a fallback to the project's node modules.  Useful for linked modules with peer dependencies.  When such a package is installed normally the peer dependencies would normally be located in the project's node modules.

Comment (filipesilva) [inline packages/@angular/cli/models/webpack-configs/common.ts]: I get it now, it's for when the linked library does not contain the peerDep but the app does. 👍

**Comment by jshcrowthe** [comment]
I've validated this fix for my use cases, and it looks great! Thanks for your help @clydin.

**Pull request merged**